protos/coinbase/chainstorage/blockchain_xrp.proto

```diff
@@ -0,0 +1,223 @@
+}
+
+message TakerGets {
+  string currency = 1; // The currency value in hexadecimal format.
+  string issuer = 2;   // The issuer of the currency, if applicable.
+  string value = 3;    // The actual value of the currency.
+}
+
+// DeletedNode contains information about a deleted node.
+message DeletedNode {
+  message FinalFields {
+    string account = 1;       // The account related to the node.
+    string book_directory = 2; // The book directory of the node.
+    string book_node = 3;      // The book node of the node.
+    uint32 flags = 4;            // The flags associated with the node.
+    string owner_node = 5;     // The owner node of the node.
+    string previous_txn_id = 6; // The transaction ID of the previous transaction.
+    uint32 previous_txn_lgr_seq = 7; // The ledger sequence of the previous transaction.
+    uint32 sequence = 8;        // The sequence number of the node.
+    TakerGets taker_gets = 9;    // The taker gets value of the node.
+    string taker_pays = 10;    // The taker pays value of the node.
+  }
+  string ledger_entry_type = 11; // The type of ledger entry.
+  string ledger_index = 12;     // The ledger index of the node.
+  FinalFields final_fields = 13; // The final fields of the node.
+}
+
+// ModifiedNode contains information about a modified node.
+message ModifiedNode {
+  message FinalFields {
+    uint32 flags = 1;           // The flags associated with the node.
+    string index_next = 2;       // The next index of the node.
+    string index_previous = 3;   // The previous index of the node.
+    string owner = 4;            // The owner of the node.
+    string root_index = 5;       // The root index of the node.
+  }
+  string ledger_entry_type = 6; // The type of ledger entry.
+  string ledger_index = 7;     // The ledger index of the node.
+  FinalFields final_fields = 8; // The final fields of the node.
+  FinalFields previous_fields = 9; // The previous fields of the node.
+  string previous_txn_id = 10; // The transaction ID of the previous transaction.
+  uint32 previous_txn_lgr_seq = 11; // The ledger sequence of the previous transaction.
+}
+
+// CreatedNode contains information about a created node.
+message CreatedNode {
+  string ledger_entry_type = 1; // The type of ledger entry.
+  string ledger_index = 2;       // The ledger index of the node.
+  message NewFields {
+    string exchange_rate = 1; // The exchange rate of the node.
+    string root_index = 2;      // The root index of the node.
+    string taker_gets_currency = 3; // The currency that the taker gets.
+    string taker_gets_issuer = 4;  // The issuer of the currency that the taker gets.
+    string taker_pays_currency = 5;  // The currency that the taker pays.
+    string taker_pays_issuer = 6;   // The issuer of the currency that the taker pays.
+  }
+  NewFields new_fields = 3; // The new fields of the node.
+}
+
+// TxResult represents the result of a transaction query.
+message LedgerTransaction {
```

Comment:
can you double check that these fields covers all the different transaction types here https://xrpl.org/docs/references/protocol/transactions/types/.  It's also not clear to me how Pseudo-Transactions are represented in the API.